includes/admin/extensions/class-extensions-api.php

```diff
@@ -0,0 +1,169 @@
+			$array_key = ! empty( $item_id ) ? $item_id : $body[ $key ];
+			update_option(
+				$option_name,
+				array(
+					$array_key => false,
+					'timeout'  => strtotime( '+1 hour', time() ),
+				),
+				false
+			);
+
+			return false;
+		}
+
+		$response = json_decode( wp_remote_retrieve_body( $request ) );
+		$value    = array(
+			'timeout' => strtotime( '+1 week', time() ),
+		);
+		if ( $item_id && ! empty( $response->$item_id ) ) {
+			$item              = $response->$item_id;
+			$value[ $item_id ] = $this->get_item_data( $item );
+		} elseif ( in_array( $key, array( 'category', 'tag' ), true ) ) {
+			$term_id = $body[ $key ];
+			foreach ( $response as $item_id => $item ) {
+				if ( 'category' === $key && ( empty( $item->categories ) || ! in_array( $term_id, $item->categories, true ) ) ) {
+					continue;
+				} elseif ( 'tag' === $key && ( empty( $item->tags ) || ! in_array( $term_id, $item->tags, true ) ) ) {
+					continue;
+				}
+				$value[ $item_id ] = $this->get_item_data( $item );
+			}
+		}
+
+		update_option( $option_name, $value, false );
+
+		return $item_id && ! empty( $value[ $item_id ] ) ? $value[ $item_id ] : $value;
+	}
+
+	/**
+	 * Gets the product data as needed for the extension manager.
+	 *
+	 * @since 2.11.x
+	 * @param object $item
+	 * @return array
+	 */
+	private function get_item_data( $item ) {
+		return array(
+			'title'       => ! empty( $item->title ) ? $item->title : '',
+			'slug'        => ! empty( $item->slug ) ? $item->slug : '',
+			'image'       => ! empty( $item->image ) ? $item->image : '',
+			'description' => ! empty( $item->excerpt ) ? $item->excerpt : '',
+			'basename'    => ! empty( $item->custom_meta->basename ) ? $item->custom_meta->basename : '',
+			'tab'         => ! empty( $item->custom_meta->settings_tab ) ? $item->custom_meta->settings_tab : '',
+			'section'     => ! empty( $item->custom_meta->settings_section ) ? $item->custom_meta->settings_section : '',
+		);
+	}
+
+	/**
+	 * Gets the stored extension data from the database.
+	 * If it doesn't exist, or has expired, deletes the option and returns false.
+	 *
+	 * @since 2.11.x
+	 * @param string $option_name The option name to look for in the database.
+	 * @return array|bool         Returns the option data if not expired, or false if expired or doesn't exist yet.
+	 */
+	private function get_stored_extension_data( $option_name ) {
+		$option = get_option( $option_name );
+		if ( ! empty( $option['timeout'] ) && time() <= $option['timeout'] ) {
+			return $option;
+		};
+
+		delete_option( $option_name );
+		return false;
+	}
+
+	/**
+	 * Gets the EDD REST API URL for products.
+	 *
+	 * @since 2.11.x
+	 * @return string
+	 */
+	private function get_products_api_url() {
+		if ( defined( 'EDD_PRODUCTS_API_URL' ) ) {
+			return EDD_PRODUCTS_API_URL;
+		}
+
+		return 'https://easydigitaldownloads.com/wp-content/uploads/cache/products/products.json';
```

Comment:
Can we adjust this to be the query stringed version?

`https://easydigitaldownloads.com/?edd_action=extension_data`? That way we can adjust the location of the `.json` file if needed or naming, without having to maintain backwards compatibility. They should both return the same data set.